### PR TITLE
Add support for batching metrics in Client.

### PR DESCRIPTION
**Pull request by tixxit**
Need to add some tests, but the basic idea is here and existing tests are passing.

### CODE REVIEW

File: src/main/scala/github/gphat/censorinus/Client.scala [hunk 1]
```diff
@@ -115,11 +147,90 @@ class Client(
...
+     * the duration of `f`. Any use of the `ByteBuffer` outside of this will
+     * result in undefined and, likely, confusing/incorrect behaviour. This
+     * will completely exhaust `line`.
+     *
+     * @param lines the encoded metrics to batch
+     * @param f the caller-supplied function to send the batched metrics to
+     */
+    def batch(lines: Iterator[String])(f: ByteBuffer => Unit): Unit = {
+      val buffered = lines.buffered
+      while (lines.hasNext) {
```
Comment (oscar-stripe) [inline src/main/scala/github/gphat/censorinus/Client.scala]: why `while(lines.hasNext) {` and not `buffered.hasNext`?

Comment (tixxit) [reply src/main/scala/github/gphat/censorinus/Client.scala]: Definitely a typo!

**Comment by gphat** [comment]
I am not familiar with the ByteBuffer, Encoder and CoderResult classes but upon reading up on them, all seems legit. I was pretty interested in how you'd do this. I likely wouldn't have used as much interesting FP stuff as you did, so this was a fun read! Lemme know when you're ready to look at releasing. (I'm OoO until Monday tho!)

**Comment by tixxit** [comment]
I've added tests and un-WIP'd this! Feel free to release this whenever you get back - no big rush.

**Pull request merged**